### ISSUE TITLE
Can't use FillImputer with @pipeline ?

### ISSUE THREAD

**Issue by onetonfoot** [alan-turing-institute/MLJ.jl#320]
`FillImputer` doesn't work with the `@pipeline`  macro
```julia
data = DataFrame(:x1 => [1,2,3,missing,4]);

@pipeline Filler(
    filla = FillImputer()
)

mach = machine(D(), data)
fit!(mach)

# TypeError: non-boolean (Missing) used in boolean context
```

It's not the end of the world since it works fine outside of `@pipeline`
however it was definitely unexpected since I'd think something like this would work fine 
```julia
 @pipeline Preprocess(
    fill = FillImputer(),
    std = Standardizer()
)
```

**Comment by ablaom** [comment]
@onetonfoot Great catch. Thanks for posting.

Turns out to be a subtle issue with "in" for objects of MLJType. With this fixed, we get:

```julia
julia> data = DataFrame(:x1 => [1,2,3,missing,4]);
julia> filler = @pipeline Filler(
       filla = FillImputer()
       )
julia> mach = machine(filler, data)
julia> fit!(mach)
[ Info: Training Machine{Filler} @ 3…74.
[ Info: Training NodalMachine{FillImputer} @ 1…17.
Machine{Filler} @ 3…74

julia> transform(mach, data)
5×1 DataFrame
│ Row │ x1    │
│     │ Int64 │
├─────┼───────┤
│ 1   │ 1     │
│ 2   │ 2     │
│ 3   │ 3     │
│ 4   │ 2     │
│ 5   │ 4     │
```

Working on PR to resolve the issue in MLJBase now.

**Comment by ablaom** [comment]
Should be resolved by MLJBase 0.7.4.
